File: EmployeeApplication/src/main/java/application/schema/Order.java
package application.schema;

import java.util.List;

/**
 * Order Schema
 *
 * @param _id
 * @param customerEmail
 * @param price
 * @param type
 * @param state
 * @param items
 */
public record Order(
    String _id,
    String customerEmail,
    int price,
    String type,
    String state,
    List<OrderItem> items) {}

File: server/src/schema/order.ts
export interface Order {
	customerEmail: string;
	price: number;
	type: string;
	items: OrderItem[];
}

export interface OrderItem {
	item: ItemId;
	quantity: number;
}

export interface ItemId {
	name: string
}

export interface UpdateOrder {
	_id: string,
	state: string
}
